tests/vpp.cpp

```diff
@@ -133,39 +148,138 @@ class VppTest
         return true;
     }
 private:
+    bool processCmdLine(int argc, char* argv[])
+    {
+        char opt;
+        const struct option long_opts[] = {
+            { "help", no_argument, NULL, 'h' },
+            { "sharpening", required_argument, NULL, 's' },
+            { "dn", required_argument, NULL, 0 },
+            { "di", required_argument, NULL, 0 },
+            { NULL, no_argument, NULL, 0 }
+        };
+        int option_index;
+
+        if (argc < 3) {
+            usage();
+            return false;
+        }
+
+        while ((opt = getopt_long_only(argc, argv, "s:h:", long_opts, &option_index)) != -1) {
+            switch (opt) {
+            case 'h':
+            case '?':
+                usage();
+                return false;
+            case 's':
+                m_sharpening = atoi(optarg);
+                break;
+            case 0:
+                switch (option_index) {
+                case 2:
+                    m_denoise = atoi(optarg);
+                    break;
+                case 3:
+                    m_deinterlaceMode = optarg;
+                    break;
+                default:
+                    usage();
+                    return false;
+                }
+                break;
+            default:
+                usage();
+                return false;
+            }
+        }
+        if (optind + 2 < argc) {
+            usage();
+            return false;
+        }
+        m_inputName = argv[optind++];
+        m_outputName = argv[optind++];
+        return true;
+    }
+
     bool createVpp()
     {
         NativeDisplay nativeDisplay;
         nativeDisplay.type = NATIVE_DISPLAY_VA;
         nativeDisplay.handle = (intptr_t)*m_display;
         m_vpp.reset(createVideoPostProcess(YAMI_VPP_SCALER), releaseVideoPostProcess);
-        return m_vpp->setNativeDisplay(nativeDisplay) == YAMI_SUCCESS;
+        if (m_vpp->setNativeDisplay(nativeDisplay) != YAMI_SUCCESS)
+            return false;
+#if YAMI_CHECK_API_VERSION(0, 2, 2)
+        {
```

Comment:
It's intend to do. There are similar block below, Add brace to avoid `VPPDenoiseParameters denoise; `reference by other block